ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/productcarousel/v1/productcarousel/clientlibs/js/product-carousel.js

```diff
@@ -0,0 +1,121 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+(function () {
+
+	/**
+        *	Product Carousel Component	
+	*/    
+    class ProductCarousel {
+
+		// Initialise the variables
+      constructor() {
+		this._speed             = 300;
+        this._delay             = 0;
+        this._effect            = 'linear';  
+        this._product_carousel_parent = document.querySelector('.product-carousel-parent');
+        this._cardsContainer = document.querySelector('.cards-container'); 
+        this._cards = this._cardsContainer.querySelectorAll('.product-card');
+        this._btnPrev = document.querySelector('#prev-btn');
```

Comment:
Please don't use IDs for DOM elements, for the exact same reason @herzog31 mentioned above - two carousel elements in the page. Another reason would be semantics - we want to say that this button has a specific role to fulfill (move the carousel) so we should use data attributes.  Once you have a root node passed in the constructor you can always do something like `rootNode.querySelector("button[data-carousel-action='next']")`

Reply:
data attribute is picked to refer the element